src/routes/guilds/guild/leaderboard.ts

```diff
@@ -0,0 +1,80 @@
+import { Route, RouteStore } from 'klasa-dashboard-hooks';
+import ApiRequest from '../../../lib/structures/api/ApiRequest';
+import ApiResponse from '../../../lib/structures/api/ApiResponse';
+import { Snowflake } from 'discord.js';
+import { LeaderboardUser } from '../../../lib/util/Leaderboard';
+
+export default class extends Route {
+
+	public constructor(store: RouteStore, file: string[], directory: string) {
+		super(store, file, directory, { route: 'guilds/:guild/leaderboard' });
+	}
+
+	public async get(request: ApiRequest, response: ApiResponse) {
+
+		const guildID = request.params.guild;
+		if (!this.client.guilds.has(guildID)) return response.error(400);
+		const lb = await this.client.leaderboard.fetch(guildID);
+
+		if ((Object.keys(request.query)).includes('id')) {
+
+			const id = request.query.id as string;
+			const lbData = lb.get(id);
+			if (!lbData) return response.error(400);
+
+			lbData.name = lbData.name ? lbData.name : (await this.client.users.fetch(id)).username;
+			return response.status(200).json({ id, ...lbData });
+		} else if ((Object.keys(request.query)).includes('position')) {
+
+			const id = lb.findKey(e => e.position === Number(request.query.position));
+			if (!id) return response.error(400);
+
+			const lbData = lb.get(id) as LeaderboardUser;
+			lbData.name = lbData.name ? lbData.name : (await this.client.users.fetch(request.query.id as string)).username;
+
+			return response.status(200).json({ id, ...lbData });
+		}
+
+		const limit = (Object.keys(request.query)).includes('limit') ? Number(request.query.limit) : 10;
+		const before = (Object.keys(request.query)).includes('before') ? Number(request.query.before) - 1 : undefined;
+		const after = ((Object.keys(request.query)).includes('after') ? Number(request.query.after) + 1 : 1);
+
+		if (isNaN(limit) || limit! > 1000) return response.error(400);
+		if ((before && (before! < 0)) || (before && isNaN(before))) return response.error(400);
+		if (isNaN(after)) return response.error(400);
```

Comment:
You should use `isNumber` as imported from `@klasa/utils`.

Reply:
but... why is that? AFAIK importing code makes the code execute slower, doesn't it?

Reply:
You have a very terribly bad assumption... starting that `isNaN` has more overhead (as it re-parses the number) than `Number.isNaN` (which takes strictly a number), but either way we need to ensure those are completely correct numbers.

I mean, either do that or use `Number.isInteger`, which takes into account everything from `isNumber` and `isNaN`.

Reply:
alrighty, thanks